Feature/559 authorize notifications

solita-pekko opened the pull request:
Start asking users to enable notifications in preparation of the app starting to display notifications about exposures itself.

solita-pekko committed:
Added asking permission to display notifications.

solita-pekko committed:
Fix radar status being incorrect after onboarding.

If user didn't grant permission to display notifications, then
radar status was "on" after onboarding had completed (radar status had
been set to "notificationsOff" before the step that enables
the use of EN API).

solita-pekko committed:
Update CHANGELOG.md

solita-pekko committed:
Updated English translations.

Commit:
Refactoring

- Instead of a global Notifications object, use a protocol to hide the implementation to make mocking easier in ExposureRepository tests and provide the new NotificationService through Environment
- Remove redundant callback from ExposureRepository.refreshStatus()
- Swedish translation
- Move StatusHeaderView button logic from the view to the MainViewController by using a delegate
- Add a scroll view to the test UI screen

Commit:
Revert redundant mutability

spheroid reviewed (commented):
Minor refactoring, and while I was at it, I also refactored the StatusHeaderView to use a delegate for button logic.

Commit:
Text changes

TomiLahtinen committed:
Refactored Notification Center delegate implementation from AppDelegate to SceneDelegate

TomiLahtinen commented on Koronavilkku/Koronavilkku/Tools/NotificationService.swift:
```diff
@@ -0,0 +1,121 @@
+        }
+    }
+    
+    func showExposureNotification(exposureCount: Int?, delay: TimeInterval? = nil) {
```
Maybe move this to exposure repository

spheroid replied:
Done

TomiLahtinen committed:
Minor refactoring

TomiLahtinen commented on Koronavilkku/Koronavilkku/ViewController/SettingsViewController.swift:
```diff
@@ -117,7 +117,7 @@ class SettingsViewController: UIViewController {
         
         switch LocalStore.shared.uiStatus {
-        case .on:
+        case .on, .notificationsOff:
```
Works here but when navigating to next view, it renders false information

spheroid replied:
Has been fixed in ChangeRadarStatusViewController as well

TomiLahtinen replied:
Settings view seems to reflect properly all app states now

Commit:
Fix review issues

- Move exposure-related notification logic to ExposureRepository
- Fix enabled state in Settings when notifications are not authorized

Commit:
Remove redundant mutability

Commit:
Update CHANGELOG.md
